src/utils/generatePKCE.ts

```diff
@@ -0,0 +1,33 @@
+/*
+Utility functions for PCKE
+*/
+import crypto from "crypto";
+
+export type CodeChallenge = {
+  codeChallenge: string;
+  verifier: string;
+};
+
+function base64URLEncode(buffer: Buffer): string {
+  return buffer
+    .toString("base64")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=/g, "");
+}
+
+function sha256(str: string): Buffer {
+  return crypto.createHash("sha256").update(str).digest();
+}
+
+export function generateCodeChallenge(): CodeChallenge {
```

Comment:
I know this is basically just lifted from the sample app but I feel like it's named a little weird. Just because it turns out to be: `codeChallenge.codeChallenge` sort of thing... So maybe consider a different name for the wrapper around verifier + codeChallenge? Maybe something like pkceData? I dunno...